SAK-42331 WIP move local styles from delegatedaccess to library

profmikegreene opened the pull request:
https://jira.sakaiproject.org/browse/SAK-42331

Since this is a jquery plugin, I created a library/sass/lib folder modeled after the library/js/lib structure instead of adding it into the modules folder.

profmikegreene commented on library/src/webapp/js/headscripts.js:
```diff
@@ -782,6 +782,9 @@ function includeWebjarLibrary(library) {
     document.write('\x3Clink rel="stylesheet" href="' + webjars + 'jquery.tablesorter/' + libraryVersion + '/dist/css/jquery.tablesorter.pager.min.css' + ver + '"/>');
+	} else if (library === "jquery.asmselect") {
+		libraryVersion = "1.0.4a-beta";
+		document.write('\x3Clink rel="stylesheet" href="/library/js/jquery/asmselect/' + libraryVersion + '/jquery.asmselect.css' + ver + '"/>');
```
this doesn't seem right to me but I don't know how to make the file appear under the webjars directory vs webapps/library/js during the deploy process.

Do these libraries have to be in webjars.org? I tried adding
```xml
<!--Used in Delegated Access, Site Manage -->
        <dependency>
            <groupId>org.webjars</groupId>
            <artifactId>asmselect</artifactId>
            <version>1.0.4a-beta</version>
            <scope>runtime</scope>
        </dependency>
```
to the POM but that created a build error and I couldn't find asmselect on webjars.org.

jonespm commented:
It sounds like this library (asmselect 11 years old) should just be replaced with something newer.
The code was mirrored from
https://github.com/weplay/jquery-asmselect

And there is this one that's also a little old
https://github.com/vicb/bsmSelect

It looks like Bootstrap Multiselect might be the same as this: https://github.com/davidstutz/bootstrap-multiselect

There might be other Bootstrap or Multiselect libraries out there too.

ottenhoff commented:
I believe @mpellicer has researched multi-select quite a bit

jonespm commented:
Yeah right, I Migeul had switched to that bootstrap-multiselect then to select2. So this package left in delegated access just got left behind.

mpellicer commented:
Exactly @jonespm and @ottenhoff,

We replaced many selects by the boostrap-multiselect searching for consistency:
https://github.com/sakaiproject/sakai/blob/master/library/pom.xml#L233

Select2 is also used in Samigo or Private messages. 
https://github.com/sakaiproject/sakai/blob/master/library/pom.xml#L212

When I wanted to condense and choose only one, some folks said "boostrap-multiselect" has more features than select2 so having both could be a good approach.

Miguel

profmikegreene commented:
I did more work as I kept finding issues

1. replaced the asmselect library with bootstrap-multiselect
2. SWITCH-ified the sak-banner markup
3. addressed some accessibility concerns that accesslint brought up
4. replaced the table-based layout with bootstrap grid

I think there's a commit for each of those items if we don't want to take one of them. I can't seem to get all the accesslint errors to go away, is that a general issue between accesslint and wicket?
